chapters/depth.md

```diff
@@ -0,0 +1,260 @@
+
+### Depth Range
+
+Each viewport holds a `VkViewport::minDepth` and `VkViewport::maxDepth` value which sets the "depth range" for the viewport.
+
+> Note: Despite their names, `minDepth` can be less than, equal to, or greater than `maxDepth`.
+
+The `minDepth` and `maxDepth` are restricted to be set inclusively between `0.0` and `1.0`. If the [VK_EXT_depth_range_unrestricted](https://www.khronos.org/registry/vulkan/specs/1.2-extensions/man/html/VK_EXT_depth_range_unrestricted.html) is enabled, this restriction goes away.
+
+The framebuffer depth coordinate `Zf` is represented as:
+
+`Zf` = `Pz` * `Zd` * `Oz`
+- `Zd` = `Zc`/`Wc` (see Primitive Clipping)
+- `Oz` = `minDepth`
+- `Pz` = `maxDepth` - `minDepth`
+
+# Rasterization
+
+## Depth Bias
+
+The depth values of all fragments generated by the rasterization of a polygon can be offset by a single value that is computed for that polygon. If `VkPipelineRasterizationStateCreateInfo::depthBiasEnable` is `VK_FALSE` at draw time, no depth bias is applied.
+
+Using the `depthBiasConstantFactor`, `depthBiasClamp`, and `depthBiasSlopeFactor` in `VkPipelineRasterizationStateCreateInfo` the depth bias [can be calculated](https://www.khronos.org/registry/vulkan/specs/1.2-extensions/html/vkspec.html#primsrast-depthbias).
+
+> Requires the `VkPhysicalDeviceFeatures::depthBiasClamp` feature to be supported otherwise `VkPipelineRasterizationStateCreateInfo::depthBiasClamp` must be `0.0f`.
+
+> Note: The depth bias values can be set [dynamically](dynamic_state.md) using `VK_DYNAMIC_STATE_DEPTH_BIAS` or the `VK_DYNAMIC_STATE_DEPTH_BIAS_ENABLE_EXT` from [VK_EXT_extended_dynamic_state2](https://www.khronos.org/registry/vulkan/specs/1.2-extensions/man/html/VK_EXT_extended_dynamic_state2.html).
+
+# Post-rasterization
+
+## Fragment Shader
+
+The input built-in `FragCoord` is the framebuffer coordinate. The `Z` component is the interpolated depth value of the primitive. This `Z` component value will be written to `FragDepth` if the shader doesn't write to it. If the shader dynamically writes to `FragDepth`, the `DepthReplacing` Execution Mode must be declared (This is done in tools such as glslang).
+
+> `FragDepth` and `FragCoord` are `gl_FragDepth` and `gl_FragCoord` in GLSL
+
+> Note: When using `OpTypeImage` in SPIR-V the `Depth` operand is ignored in Vulkan
+
+### Conservative depth
+
+The `DepthGreater`, `DepthLess`, and `DepthUnchanged` Executation Mode allow for a possible optimization for implementations that [relies on an early depth test to be run before the fragment](https://www.khronos.org/registry/OpenGL/extensions/ARB/ARB_conservative_depth.txt). This can be easily done in GLSL by declaring `gl_FragDepth` with the proper layout qualifier.
```

Comment:
```suggestion
The `DepthGreater`, `DepthLess`, and `DepthUnchanged` Execution Mode allow for a possible optimization for implementations that [relies on an early depth test to be run before the fragment](https://www.khronos.org/registry/OpenGL/extensions/ARB/ARB_conservative_depth.txt). This can be easily done in GLSL by declaring `gl_FragDepth` with the proper layout qualifier.
```